integration/yarn/src/test/java/tachyon/yarn/ApplicationMasterTest.java

```diff
@@ -229,11 +238,73 @@ public Void answer(InvocationOnMock invocation) throws Throwable {
    */
   @Test(timeout = 10000)
   public void negotiateUniqueWorkerHostsTest() throws Exception {
+    mockResourceManager(NUM_WORKERS);
+
+    // Wait for all workers to be allocated, then shut down mMaster
+    new Thread(new Runnable() {
```

Comment:
define this thread in a util function? because it is used unmodified multiple times.